modules/examples/src/main/scala/iotaexamples/NestedCoproducts.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016-2017 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package iotaexamples
+
+import iota._
+
+import scala.Predef._
+
+object NestedCoproducts extends App {
+
+  case class Op0[A]()
+  case class Op1[A]()
+  case class Op2[A]()
+  case class Op3[A]()
+  case class Op4[A]()
+  case class Op5[A]()
+
+  import KList.::
+  import KList.Op.Concat
+  import KList.Op.Reverse
+
+  type AlgA[A] = CopK[Op0 :: Op1 :: Op2 :: KNil, A]
+  type AlgB[A] = CopK[Op3 :: Op4 :: Op5 :: KNil, A]
+
+  type AFwd[A] = CopK[Concat[AlgA[_]#L, AlgB[_]#L], A]
```

Comment:
This _should_ allow us to address https://github.com/frees-io/freestyle/issues/322